src/main/java/GameHistoryWriter.java

```diff
@@ -0,0 +1,42 @@
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class GameHistoryWriter {
+
+    DateTimeParser dateTimeParser;
+    BufferedWriter out;
+    FileWriter fWriter;
+
+    public GameHistoryWriter(DateTimeParser dateTimeParser) throws IOException {
+        this.dateTimeParser = dateTimeParser;
+        this.fWriter = new FileWriter("GameHistory.txt", true);
+        this.out = new BufferedWriter(fWriter);
+
+    }
+
+    public void writeGameInfoToFile(String winner, String[] board, BufferedWriter out){
+        try {
+            out.write("DATE: " + dateTimeParser.formatCurrentDate(dateTimeParser.dateFormatter, dateTimeParser.localDate) + ", ");
```

Comment:
```suggestion
            out.write("DATE: " + dateTimeParser.formatCurrentDate() + ", ");
```
You should never ask a class for state it has just to pass that back to it.

Reply:
✅